include/fiction/algorithms/simulation/sidb/quicksim.hpp

```diff
@@ -93,7 +99,7 @@ sidb_simulation_result<Lyt> quicksim(const Lyt& lyt, const quicksim_params& ps =
         charge_lyt.assign_base_number(2);
         charge_lyt.assign_all_charge_states(sidb_charge_state::NEGATIVE);
         charge_lyt.update_after_charge_change(dependent_cell_mode::VARIABLE);
-        const auto negative_sidb_indices = charge_lyt.negative_sidb_detection();
+        const auto& negative_sidb_indices = charge_lyt.negative_sidb_detection();
```

Comment:
Why the reference? Due to C++'s return value optimization, it should be equivalent performance-wise to have `const auto` instead of `const auto&`. However, `const auto` is safer because it doesn't fall prey to the object running out of scope and disappearing. I prefer using `const auto` unless there are good reasons against it.